### PR TITLE
Accept a canvas as parameter as well

### PR DESCRIPTION
**Pull request by Prinzhorn**
I use https://github.com/nodeca/pica to resize huge images and want to rotate the result of the orientation. Accepting a canvas as parameter as well makes sense and the changes are simple enough.

I also changed the demo as I was confused why it didn't reflect my changes while I was trying them and found that it links to the version on GH.

### CODE REVIEW

File: exif-orient.js [hunk 1]
```diff
@@ -16,8 +16,8 @@
    * @return {HTMLCanvasElement} a canvas object.
    */
   return function exifOrient(img, orientation, cb) {
-    if (typeof img !== 'string' && !(img instanceof HTMLImageElement)) {
-      return cb(new Error('img must be a string or an HTMLImageElement'))
+    if (typeof img !== 'string' && !(img instanceof HTMLImageElement || img instanceof HTMLCanvasElement)) {
```
Comment (buunguyen) [inline exif-orient.js]: Easier to read if not mixing & and |:

``` js
if (typeof img !== 'string' && !(img instanceof HTMLImageElement) && !(img instanceof HTMLCanvasElement))
```

File: exif-orient.js [hunk 2]
```diff
@@ -16,8 +16,8 @@
    * @return {HTMLCanvasElement} a canvas object.
```
Comment (buunguyen) [inline exif-orient.js:16]: Need to update comment, currently {HTMLImageElement | String}

File: demo/index.html [hunk 3]
```diff
@@ -14,12 +14,12 @@
     }
   </style>
   <script src="http://rawgit.com/exif-js/exif-js/master/exif.js"></script>
-  <script src="http://rawgit.com/buunguyen/exif-orient/master/exif-orient.js"></script>
+  <script src="../exif-orient.js"></script>
```
Comment (buunguyen) [inline demo/index.html:17]: Have you tested that this change work when accessed via http://rawgit.com/YOUR-NAME/exif-orient/master/demo/index.html?

**Review by buunguyen — CHANGES_REQUESTED** [review]

File: demo/index.html (hunk 3 above)
Comment (Prinzhorn) [reply demo/index.html:17]: See http://rawgit.com/prinzhorn/exif-orient/accept-canvas/demo/index.html

**Comment by Prinzhorn** [comment]
I updated the if statement and the comment

**Pull request merged**